src/utils/theme.ts

```diff
@@ -0,0 +1,39 @@
+import { autoTheme, darkTheme } from "./resources";
+
+/**
+ * Emits when the theme is dynamically toggled between light and dark on <body>.
+ * This file is imported in Stencil's `globalScript` config option.
+ *
+ * @see {@link https://stenciljs.com/docs/config#globalscript Stencil's globalScript property}
+ */
+export default function (): void {
```

Comment:
Once we have a global script can you name this function to indicate it sets up theme update changes?

Reply:
You’ll also need to update this method to return early if not running on the browser, so the SSR build doesn’t go 💥.